extensions/amp-story/1.0/amp-story-tooltip.js

```diff
@@ -0,0 +1,401 @@
+
+    this.tooltipOverlayEl_ =
+      dev().assert(this.buildTemplate_(this.win_.document));
+    createShadowRootWithStyle(this.shadowRoot_, this.tooltipOverlayEl_, CSS);
+
+    this.tooltipOverlayEl_
+        .addEventListener('click', event => this.onOutsideClickableEls_(event));
+
+    this.storeService_.subscribe(StateProperty.UI_STATE, isDesktop => {
+      this.onUIStateUpdate_(isDesktop);
+    }, true /** callToInitialize */);
+
+    this.storeService_.subscribe(StateProperty.CURRENT_PAGE_ID, () => {
+      // Hide active tooltip when page switch is triggered by keyboard or
+      // desktop buttons.
+      this.closeTooltip_();
+    });
+
+    return this.shadowRoot_;
+  }
+
+  /**
+   * Hides the tooltip layer.
+   * @private
+   */
+  closeTooltip_() {
+    this.storeService_.dispatch(Action.TOGGLE_TOOLTIP, null);
+    // Hide active tooltip when page switch is triggered by keyboard or
+    // desktop buttons.
+    this.onTooltipStateUpdate_(false);
+  }
+
+  /**
+   * Reacts to store updates related to tooltip active status.
+   * @param {boolean} tooltipIsActive
+   * @private
+   */
+  onTooltipStateUpdate_(tooltipIsActive) {
+    this.resources_.mutateElement(dev().assertElement(this.tooltipOverlayEl_),
+        () => {
+          this.tooltipOverlayEl_
+              .classList.toggle('i-amphtml-hidden', !tooltipIsActive);
+        });
+  }
+
+  /**
+   * Reacts to desktop state updates and hides navigation buttons since we
+   * already have in the desktop UI.
+   * @param {!UIType} uiState
+   * @private
+   */
+  onUIStateUpdate_(uiState) {
+    this.resources_.mutateElement(dev().assertElement(this.tooltipOverlayEl_),
+        () => {
+          uiState === UIType.DESKTOP ?
+            this.tooltipOverlayEl_.setAttribute('desktop', '') :
+            this.tooltipOverlayEl_.removeAttribute('desktop');
+        });
+  }
+
+  /**
+   * Builds tooltip and attaches it depending on the clicked element content and
+   * position.
+   * @param {!Element} clickedEl
+   * @private
+   */
+  attachTooltipToEl_(clickedEl) {
+    const {href} = parseUrlDeprecated(clickedEl.getAttribute('href'));
+
+    const domainName = getSourceOriginForElement(clickedEl, href);
+    const tooltipText = clickedEl.getAttribute('tooltipText') || domainName;
+    this.appendTextToTooltip_(tooltipText);
+
+    const iconAttr = clickedEl.getAttribute('tooltipIcon');
+    const iconSrc = iconAttr ? parseUrlDeprecated(iconAttr).href : '';
+    this.appendIconToTooltip_(iconSrc);
+
+    this.positionTooltip_(clickedEl);
+  }
+
+  /**
+   * Checks for existing text content and modifies it or appends a new text node
+   * if building tooltip for the first time.
+   * @param {string} tooltipText
+   * @private
+   */
+  appendTextToTooltip_(tooltipText) {
+    const existingTooltipText = childElement(
+        dev().assertElement(this.tooltip_),
+        el => el.classList.contains('i-amphtml-tooltip-text'));
```

Comment:
Is there any reason you have to do this over a simple `this.tooltip_.querySelector(.i-amphtml-tooltip-text)`?